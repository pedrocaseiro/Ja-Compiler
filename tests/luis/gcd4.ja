class gcd4 {
	public static double a;

	public static void main(boolean b, double a1) {
		//System.out.println(a);
		System.out.println(zxc);

		//main(3, 3);
		//asd(3, false);

		boolean x;
		x = true;
		//asd(3, x);

		if(x*3)

		System.out.println("haha");
		printOne();
		a = 4+3;
		return 3+3.5;
	}

	public static void printOne() {
    	System.out.println("Hello World");
    	main(false, 3.14);
  	}

	public static int main (int x1, boolean z) {
		printOne();
		/*int x;
		int y;
		x = 3;
		y = 4;
		main(3, 3.5);
		if (x == 3.5) {

		}
		k = 3.4;*/
	}
}
