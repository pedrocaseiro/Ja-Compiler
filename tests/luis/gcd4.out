Line 6, col 22: Cannot find symbol zxc
Line 6, col 22: Incompatible type undef in System.out.println statement
Line 15, col 7: Operator * cannot be applied to types boolean, int
Line 15, col 7: Incompatible type undef in if statement
Line 20, col 11: Incompatible type double in return statement
===== Class gcd4 Symbol Table =====
a		double
main	(boolean,double)	void
printOne	()	void
main	(int,boolean)	int

===== Method main(boolean,double) Symbol Table =====
return		void
b		boolean	param
a1		double	param
x		boolean

===== Method printOne() Symbol Table =====
return		void

===== Method main(int,boolean) Symbol Table =====
return		int
x1		int	param
z		boolean	param

Program
..Id(gcd4)
..FieldDecl
....Double
....Id(a)
..MethodDecl
....MethodHeader
......Void
......Id(main)
......MethodParams
........ParamDecl
..........Bool
..........Id(b)
........ParamDecl
..........Double
..........Id(a1)
....MethodBody
......Print
........Id(zxc) - undef
......VarDecl
........Bool
........Id(x)
......Assign - boolean
........Id(x) - boolean
........BoolLit(true) - boolean
......If
........Mul - undef
..........Id(x) - boolean
..........DecLit(3) - int
........Print
..........StrLit("haha") - String
........Block
......Call - void
........Id(printOne) - ()
......Assign - double
........Id(a) - double
........Add - int
..........DecLit(4) - int
..........DecLit(3) - int
......Return
........Add - double
..........DecLit(3) - int
..........RealLit(3.5) - double
..MethodDecl
....MethodHeader
......Void
......Id(printOne)
......MethodParams
....MethodBody
......Print
........StrLit("Hello World") - String
......Call - void
........Id(main) - (boolean,double)
........BoolLit(false) - boolean
........RealLit(3.14) - double
..MethodDecl
....MethodHeader
......Int
......Id(main)
......MethodParams
........ParamDecl
..........Int
..........Id(x1)
........ParamDecl
..........Bool
..........Id(z)
....MethodBody
......Call - void
........Id(printOne) - ()
